app/src/pages/trace/TracePage.tsx

```diff
@@ -153,73 +155,77 @@ const defaultCardProps: Partial<CardProps> = {
+                span: node {
+                  context {
+                    spanId
+                  }
+                  name
+                  spanKind
+                  statusCode: propagatedStatusCode
+                  statusMessage
+                  startTime
+                  parentId
+                  latencyMs
+                  tokenCountTotal
+                  tokenCountPrompt
+                  tokenCountCompletion
+                  input {
+                    value
+                    mimeType
+                  }
+                  output {
+                    value
+                    mimeType
+                  }
+                  attributes
+                  events {
+                    name
+                    message
+                    timestamp
+                  }
+                  spanEvaluations {
+                    name
+                    label
+                    score
+                  }
+                  documentRetrievalMetrics {
+                    evaluationName
+                    ndcg
+                    precision
+                    hit
+                  }
+                  documentEvaluations {
+                    documentPosition
+                    name
+                    label
+                    score
+                    explanation
+                  }
+                  ...SpanEvaluationsTable_evals
+                }
               }
-              name
-              spanKind
-              statusCode: propagatedStatusCode
-              statusMessage
-              startTime
-              parentId
-              latencyMs
-              tokenCountTotal
-              tokenCountPrompt
-              tokenCountCompletion
-              input {
-                value
-                mimeType
-              }
-              output {
-                value
-                mimeType
-              }
-              attributes
-              events {
-                name
-                message
-                timestamp
-              }
-              spanEvaluations {
-                name
-                label
-                score
-              }
-              documentRetrievalMetrics {
-                evaluationName
-                ndcg
-                precision
-                hit
-              }
-              documentEvaluations {
-                documentPosition
-                name
-                label
-                score
-                explanation
-              }
-              ...SpanEvaluationsTable_evals
             }
           }
         }
       }
     `,
-    { traceId: traceId as string },
+    { traceId: traceId as string, id: projectId as string },
     {
       fetchPolicy: "store-and-network",
     }
   );
-  const spansList = data.spans.edges.map((edge) => edge.span);
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const spansList = data.project.spans!.edges.map((edge) => edge.span);
```

Comment:
```suggestion
const spansList = useMemo(() => {
    const gqlSpans =
      data.project.spans || ([] as NonNullable<typeof data.project.spans>);
    return gqlSpans.edges.map((edge) => edge.span);
  }, [data]);
```

Reply:
the above is the FP approach of ensuring type safety where the array functions as a 'box' (e.g. functor). The `nullable` is then just an empty array and thus can be mapped over without any danger of erroring.